File: gameserver/src/main/java/l2s/gameserver/network/authcomm/as2gs/ChangePasswordResponse.java
package l2s.gameserver.network.authcomm.as2gs;

import io.netty.buffer.ByteBuf;
import l2s.gameserver.model.Player;
import l2s.gameserver.network.authcomm.AuthServerClientService;
import l2s.gameserver.network.authcomm.ReceivablePacket;
import l2s.gameserver.network.authcomm.vertx.AuthServerConnection;
import l2s.gameserver.network.l2.GameClient;
import l2s.gameserver.network.l2.components.CustomMessage;
import l2s.gameserver.utils.Functions;

public class ChangePasswordResponse extends ReceivablePacket {
    private final String _account;
    private final boolean _changed;

    public ChangePasswordResponse(AuthServerConnection client, ByteBuf byteBuf) {
        super(client, byteBuf);

        _account = readString(byteBuf);
        _changed = byteBuf.readInt() == 1;
    }

    @Override
    protected void runImpl(AuthServerConnection client) {
        GameClient gameClient = AuthServerClientService.INSTANCE.getAuthedClient(_account);
        if (gameClient == null)
            return;
        Player activeChar = gameClient.getActiveChar();
        if (activeChar == null)
            return;
        if (_changed)
            Functions.show(new CustomMessage("scripts.commands.user.password.ResultTrue"), activeChar);
        else
            Functions.show(new CustomMessage("scripts.commands.user.password.ResultFalse"), activeChar);
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/as2gs/ChangePasswordResponse.kt
package l2s.gameserver.network.authcomm.as2gs

import io.netty.buffer.ByteBuf
import l2s.gameserver.network.authcomm.AuthServerClientService
import l2s.gameserver.network.authcomm.ReceivablePacket
import l2s.gameserver.network.authcomm.vertx.AuthServerConnection
import l2s.gameserver.network.l2.components.CustomMessage
import l2s.gameserver.network.l2.s2c.ExShowScreenMessage
import l2s.gameserver.network.l2.s2c.ExShowScreenMessage.ScreenMessageAlign

/**
 * @Author: Death
 * @Date: 8/2/2007
 * @Time: 14:39:46
 */
class ChangePasswordResponse(client: AuthServerConnection, byteBuf: ByteBuf) : ReceivablePacket(client, byteBuf) {

    private val _account: String = readString(byteBuf)
    private val _changed = byteBuf.readInt() == 1

    override fun runImpl(client: AuthServerConnection) {
        val client = AuthServerClientService.getAuthedClient(_account) ?: return
        val activeChar = client.activeChar ?: return
        if (_changed) activeChar.sendPacket(
            ExShowScreenMessage(
                CustomMessage(
                    "scripts.commands.user.password.ResultTrue"
                ).toString(activeChar),
                3000,
                ScreenMessageAlign.BOTTOM_CENTER,
                true
            )
        ) else activeChar.sendPacket(
            ExShowScreenMessage(
                CustomMessage(
                    "scripts.commands.user.password.ResultFalse"
                ).toString(activeChar),
                3000,
                ScreenMessageAlign.BOTTOM_CENTER,
                true
            )
        )
    }
}